Cargo.toml

```diff
@@ -1,3 +1,8 @@
+[profile.dev]
+# Incremental builds seem to be causing consistent compiler crashes on M1 macs
+# This was observed in version 1.58.1. Please remove if no longer relevant.
+incremental = false
```

Comment:
I actually observed the same issue on linux arm builds as well as some x86 builds randomly. But it seems to be happening on M1 builds pretty consistently. It doesn't trigger a full rebuild, it just eliminates some minor performance gains due to not recompiling each file in a crate. 
I didn't pay too much attention to how much of a difference it made in compilation times, however, it was more painful having to do a full rebuild because the incremental compilation fails and the only way to resolve is `cargo clean` and `cargo build`.

Reply:
ok. I guess we can just go with this and revisit if it does seem to have any painful repercussions.